<a href="https://colab.research.google.com/github/OpheliaVlzqz/MineriaDatos/blob/main/Avance1_PIA_10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Equipo: 10**
Integrantes


*   
1856652 Velázquez Montemayor Ofelia
*   
1856652 Velázquez Montemayor Ofelia
*
1862149 Cruz Martínez Diego Adrian
*
1867216 Martínez Reyes Jose Miguel
*
1941488 García Ortega Jose Andrés
*
1941564 Ramirez Moreno Diana Gabriela

**GRUPO 001.- Lunes de 7:00-10:00 PM**


Objetivo: Estimar una fecha en la cuál se pueda llegar a una nueva normalidad mas segura para todos, a partir de la cifras de las personas vacunadas. 


# **Base de datos principal: COVID-19 World Vaccination Progress**


URL de la pagina:  https://www.kaggle.com/datasets

# **Introducción General**

Nuestra base de datos muestra el progreso de vacunación mundial contra el COVID-19. Los coronavirus son una familia de virus que causan enfermedades (desde el resfriado común hasta enfermedades respiratorias más graves) y circulan entre humanos y animales. 
El COVID-19 es la enfermedad causada por el nuevo coronavirus conocido como SARS-CoV-2. La OMS tuvo noticia por primera vez de la existencia de este nuevo virus el 31 de diciembre de 2019, al ser informada de un grupo de casos de «neumonía vírica» que se habían declarado en Wuhan (República Popular China).

Algunos de los síntomas que puede presentar una persona con COVID-19 son, fiebre, tos seca, cansancio, pérdida del gusto y/o del olfato, dolor de garganta, dolor de cabeza, náuseas, vómito, diarrea, etc. El sector de la población con más riesgo son las personas de más de 60 años y las que padecen afecciones médicas subyacentes, como hipertensión arterial, problemas cardíacos o pulmonares, 
diabetes, obesidad o cáncer, corren un mayor riesgo de presentar cuadros graves. 
Sin embargo, cualquier persona, a cualquier edad, puede enfermar de COVID-19 y presentar un cuadro grave o morir.    
El tiempo entre la exposición a la COVID-19 y el momento en que comienzan los síntomas es, de media, de 5 o 6 días, pero puede variar entre 1 y 14 días. Por ello se recomienda que las personas que hayan estado expuestas al virus se queden en casa, alejadas de otras personas, durante 14 días, con el fin de prevenir la propagación del virus.

El COVID-19 ha afectado mucho a la humanidad, han muerto 4.49 M de personas en el mundo y se han enfermado 216 M de personas. De los casi 7,000 profesionales de la salud en el mundo que han perdido la vida por el Covid-19, al menos 1,320 se registraron en México, según un reporte de Amnistía Internacional. La pandemia ha afectado a la economía de todos los países del mundo, los precios de las materias primas cayeron, mientras subió el gasto público para la lucha contra la pandemia que  a consecuencia de esta muchos negocios cerraron, subió el índice de desempleo y aumentó la inseguridad. La deuda fiscal de muchos países se incrementó enormemente. 

Una herramienta que nos ayude a explorar los datos del COVID-19 ayudaría bastante en distintas áreas principalmente en el área médica, como por ejemplo llevar un control de las muertes, los infectados y los recuperados de las personas no vacunadas y las vacunadas, para poder ver el grado de efectividad que tienen las distintas vacunas y si tienen el mismo grado de efectividad en una persona joven o en una persona de edad avanzada. También estos datos nos ayudarían en la economía, porque con esas cifras los gobiernos sabrían cuánto riesgo tienen las personas al acudir a ciertos establecimientos y así tomar la decisión de abrirlos, cerrarlos o poner un límite de personas, ya una vez que la mayoría o una gran parte de la población se encuentra vacunada. 


# **Descripción de los datos**


**Tipo de datos**: Tablas 

**Descripción de las columnas**: la base de datos cuenta con 16 columnas y otras 4 columnas en otro archivo. 




*   País - string 
*   Código ISO del país - string
*   Fecha - object (fecha de entrada de los datos) 
Número total de vacunas - float (número absoluto total de inmunizaciones)
*   Número total de personas vacunadas - float
*   Número total de personas totalmente vacunadas - float
*   Vacunaciones diarias (en bruto) - float
*   Vacunaciones totales por cien - float (relación entre el número de vacunaciones y la población total hasta la fecha en el país)
*   Número total de personas vacunadas por cada cien - float (Relación entre la población inmunizada y la población total hasta la fecha en el país)
*   Número total de personas totalmente vacunadas por cada cien -float (Relación  entre la población totalmente inmunizada y la población total hasta la fecha en el país )
*   Número de vacunaciones por día -float
*   Vacunaciones diarias por millón - float (relación entre el número de vacunaciones y la población total para la fecha actual en el país)
*   Vacunas utilizadas en el país - string


# **Justificación del uso de datos**

Los datos se recolectaron desde el database de kaggle "COVID-19 World Vaccination Progress", el cual incluye dos archivos CSV "country_vaccinations_by_manufacturer" & "country_vaccinations"


De los cuales despreciaremos el primero, ya que la información que incluye es la cantidad de vacunas usadas de cada laboratorio en los diferentes países, y como veremos en nuestro objetivo esta información será innecesaria.


Ahora nos centraremos en la información que proporciona el database "country_vaccinations"

Algunas de las características que más nos llamaron la atención para trabajar con esta base de datos son, todos los países que se encuentran registrados en ella y cada uno de ellos tiene gran cantidad de información. Cabe destacar lo interesante de como la base desglosa la información del proceso de vacunación en cada país de una manera practica y clara, muestra en cada columna datos muy útiles para llevar un control y nos permite tener una noción de cuanta es la cantidad de inmunización de cada país

El database esta organizado de una buena manera, logrando así una facil compresnión de esta, puesto que contiene todos los países por orden alfabético y los datos de cada país están ordenados de manera cronológica, todo esto mencionado anteriormente son los beneficios que nos brinda la base de datos y así nos facilitará los siguiente análisis a realizar para cumplir el objetivo


Los datos dentro de esta base nos proporcionan la información útil puesto que, no solo tiene el numero de vacunas aplicadas diariamente, sino cuantas personas tienen por lo menos una dosis y cuales ya están completamente vacunadas y cada dato tiene su índice respecto al nivel de población lo que facilita la comparación entre países, ya que el numero de vacunas aplicadas es relativo a la población.






















 

# **Objetivo final**


Proporcionar información para reanudar las actividades cotidianas en cada uno de los países, considerado que a mayor porcentaje de población vacunada las restricciones impuestas para frenar propagación de la COVID-19 irán disminuyendo hasta desaparecer.

Para este objetivo se asume que el tipo de vacuna utilizada es igualmente efectiva y solo nos interesaremos en los porcentajes de vacunación de la población sin tomar en cuenta el tipo de vacuna.


Además de un objetivo secundario, el cual es poder analizar la información actual de cada uno de los paises para asi poder determinar cuales fueron los más eficientes en el manejo de la pandemia a nivel de vacunaciones y cuales los menos, y de esta manera poder clasificarlos según su efectividad.


# **Preguntas de interés**



*   ¿Cuál es el país que ha vacunado a mayor parte de su población?
*   ¿Qué país es el que ha tenido menor número de vacunaciones por día?




# **Limpieza de datos**

**Leer el archivo**

El primer paso sera importar las librerias las cuales podremos utilizar para la lectura y limpieza de datos. Despues se mandará llamar el archivo el cual esta en formato excel. Para eliminar los posibles simbolos que no se puedan leer utilizaremos encoding.

In [ ]:
import pandas as pd
import numpy as np
from google.colab import  files
files.upload()
covid = pd.read_csv("country_vaccinations.csv")


**Tamaño de la base de datos**

Se utiliza la función shape para revisar el tamaño de la base de datos. 

In [ ]:
covid.shape


(40204, 15)

La base de datos cuenta con 40204 filas y 15 columnas. 

Con la función head, revisaremos nuestros primeros 10 datos de la base de datos y para revisar los ultimos se utilizara la función tail 

In [ ]:
covid.head(10)

country iso_code  ...                source_name            source_website
0  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
1  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
2  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
3  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
4  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
5  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
6  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
7  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
8  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/
9  Afghanistan      AFG  ...  World Health Organization  https://covid19.who.int/

[10 rows x 15 columns]

In [ ]:
covid.tail(10)

**Manejo de columnas**

Para mejor visualización de las columnas, se decidió eliminar las columnas con información no relevante para nuestro objetivo. Para tomar esta decisión primero se mostrara el nombre de las  columnas existentes y después el tipo de dato que contienen respectivamente 


In [ ]:
covid.columns



Index(['country', 'iso_code', 'date', 'total_vaccinations',
       'people_vaccinated', 'people_fully_vaccinated',
       'daily_vaccinations_raw', 'daily_vaccinations',
       'total_vaccinations_per_hundred', 'people_vaccinated_per_hundred',
       'people_fully_vaccinated_per_hundred', 'daily_vaccinations_per_million',
       'vaccines', 'source_name', 'source_website'],
      dtype='object')

In [ ]:
covid.dtypes

country                                 object
iso_code                                object
date                                    object
total_vaccinations                     float64
people_vaccinated                      float64
people_fully_vaccinated                float64
daily_vaccinations_raw                 float64
daily_vaccinations                     float64
total_vaccinations_per_hundred         float64
people_vaccinated_per_hundred          float64
people_fully_vaccinated_per_hundred    float64
daily_vaccinations_per_million         float64
vaccines                                object
source_name                             object
source_website                          object
dtype: object

En este apartado podemos ver las columnas que no nos serán útiles para nuestro objetivo, por lo tanto serán eliminadas 


Las 4 columnas las consideramos innecesarias para nuestro objetivo ya que:


*   iso_code Es la abreviación del país, teniendo el nombre del país este dato 
es redundante
*
vaccines Es el tipo de vacuna utilizada , como solo nos interesa la cantidad de personas vacunadas sin importar el tipo este dato es innecesario
*source_name y source_website Es el nombre y la pagina de la fuente de los datos, esta información útil para saber que los datos son confiables pero no para la minería de datos




In [ ]:
covid1=covid.drop(['iso_code','source_name','source_website','vaccines'],axis=1)
covid1

country  ... daily_vaccinations_per_million
0      Afghanistan  ...                            NaN
1      Afghanistan  ...                           34.0
2      Afghanistan  ...                           34.0
3      Afghanistan  ...                           34.0
4      Afghanistan  ...                           34.0
...            ...  ...                            ...
40199     Zimbabwe  ...                         4906.0
40200     Zimbabwe  ...                         4581.0
40201     Zimbabwe  ...                         4572.0
40202     Zimbabwe  ...                         4150.0
40203     Zimbabwe  ...                         4479.0

[40204 rows x 11 columns]

En el siguiente apartado se identificarán los espacios nulos y se sustituirán por cero, para represetar que ese día no hubo datos de vacunación.

In [ ]:
covid1.isnull().any()

Pais                                                           False
Fecha                                                          False
Numero total de vacunas                                         True
Numero total de personas vacunadas                              True
Numero total de personas totalmente vacunadas                   True
Vacunaciones diarias (en bruto)                                 True
Vacunaciones diarias                                            True
Vacunaciones totales por cien                                   True
Numero total de personas vacunadas por cada cien                True
Numero total de personas totalmente vacunadas por cada cien     True
Vacunaciones diarias por millon                                 True
dtype: bool

In [ ]:
covid1.fillna(0, inplace=True)

covid1.isnull().any()

Pais                                                           False
Fecha                                                          False
Numero total de vacunas                                        False
Numero total de personas vacunadas                             False
Numero total de personas totalmente vacunadas                  False
Vacunaciones diarias (en bruto)                                False
Vacunaciones diarias                                           False
Vacunaciones totales por cien                                  False
Numero total de personas vacunadas por cada cien               False
Numero total de personas totalmente vacunadas por cada cien    False
Vacunaciones diarias por millon                                False
dtype: bool

A continuación se cambiará el nombre de las columnas, para que sea más fácil identificarlas 

In [ ]:
covid1.rename(columns={'country':'Pais'}, inplace = True)
covid1.rename(columns={'date':'Fecha'},inplace = True)
covid1.rename(columns={'total_vaccinations':'Numero total de vacunas'},inplace = True)
covid1.rename(columns={'people_vaccinated':'Numero total de personas vacunadas'},inplace = True)
covid1.rename(columns={'people_fully_vaccinated':'Numero total de personas totalmente vacunadas'},inplace = True)
covid1.rename(columns={'daily_vaccinations_raw':'Vacunaciones diarias (en bruto)'},inplace = True)
covid1.rename(columns={'daily_vaccinations':'Vacunaciones diarias'},inplace = True)
covid1.rename(columns={'total_vaccinations_per_hundred':'Vacunaciones totales por cien'},inplace = True)
covid1.rename(columns={'people_vaccinated_per_hundred':'Numero total de personas vacunadas por cada cien'},inplace = True)
covid1.rename(columns={'people_fully_vaccinated_per_hundred':'Numero total de personas totalmente vacunadas por cada cien'},inplace = True)
covid1.rename(columns={'daily_vaccinations_per_million':'Vacunaciones diarias por millon'},inplace = True)
covid1


Pais  ... Vacunaciones diarias por millon
0      Afghanistan  ...                             0.0
1      Afghanistan  ...                            34.0
2      Afghanistan  ...                            34.0
3      Afghanistan  ...                            34.0
4      Afghanistan  ...                            34.0
...            ...  ...                             ...
40199     Zimbabwe  ...                          4906.0
40200     Zimbabwe  ...                          4581.0
40201     Zimbabwe  ...                          4572.0
40202     Zimbabwe  ...                          4150.0
40203     Zimbabwe  ...                          4479.0

[40204 rows x 11 columns]


Reordenamos las columnas de orden según consideramos mas útil


In [ ]:
covid2=covid1.reindex(columns=['Pais','Fecha','Vacunaciones diarias (en bruto)','Vacunaciones diarias','Numero total de vacunas','Vacunaciones totales por cien','Numero total de personas vacunadas','Numero total de personas vacunadas por cada cien','Numero total de personas totalmente vacunadas','Numero total de personas totalmente vacunadas por cada cien','Vacunaciones diarias por millon'])
covid2

Pais  ... Vacunaciones diarias por millon
0      Afghanistan  ...                             0.0
1      Afghanistan  ...                            34.0
2      Afghanistan  ...                            34.0
3      Afghanistan  ...                            34.0
4      Afghanistan  ...                            34.0
...            ...  ...                             ...
40199     Zimbabwe  ...                          4906.0
40200     Zimbabwe  ...                          4581.0
40201     Zimbabwe  ...                          4572.0
40202     Zimbabwe  ...                          4150.0
40203     Zimbabwe  ...                          4479.0

[40204 rows x 11 columns]

# Preguntas de interes
# Cuál es el país que ha vacunado a mayor parte de su población?


Utilizaremos la función gruopby para agrupar los datos por países y calcularemos el máximo de los índices de vacunación con la funcion agg

In [ ]:
grup=covid2.groupby("Pais").agg({
    "Numero total de personas vacunadas por cada cien": "max",
    "Numero total de personas totalmente vacunadas por cada cien": "max"

})
grup

Numero total de personas vacunadas por cada cien  Numero total de personas totalmente vacunadas por cada cien
Pais                                                                                                                            
Afghanistan                                                    1.93                                               0.55          
Albania                                                       28.10                                              21.18          
Algeria                                                        7.67                                               1.62          
Andorra                                                       65.99                                              52.51          
Angola                                                         3.27                                               2.15          
...                                                             ...                                                ...          
Wales                                                         74.04                                              68.50          
Wallis and Futuna                                             44.16                                              43.06          
Yemen                                                          0.98                                               0.04          
Zambia                                                         1.64                                               1.34          
Zimbabwe                                                      16.25                                              10.35          

[222 rows x 2 columns]

Despues de obtener la tabla filtrada por países , calculáremos el índice que contenga los valores máximos con la función idxmax() y max()

In [ ]:
v1=grup["Numero total de personas vacunadas por cada cien"].idxmax()
v2=grup["Numero total de personas vacunadas por cada cien"].max()
print("El pais con mayor % de personas con al menos una dosis es", v1, "con un un", v2,"%" )


El pais con mayor % de personas con al menos una dosis es Gibraltar con un un 117.31 %


In [ ]:
v3=grup["Numero total de personas totalmente vacunadas por cada cien"].idxmax()
v4=grup["Numero total de personas totalmente vacunadas por cada cien"].max()

print("El pais con mayor % de personas totalmente vacunadas es", v3, "con un un", v4,"%" )



El pais con mayor % de personas totalmente vacunadas es Gibraltar con un un 116.34 %


El resultado a esta pregunta es interesante, puesto que el país con mayor parte de su población vacunada es gibraltar, con mas de un 100%

¿pero como es esto posible?

Gibraltar es un país pequeño con alta afluencia de extranjeros
y los datos solo miden las vacunas aplicadas sin considerar si los pacientes eran locales o no , entonces podemos asumir que Gibraltar ha aplicado más vacunas que toda su población por el flujo de extranjeros



# ¿Qué país es el que ha tenido menor número de vacunaciones por día?


In [ ]:
grup1=covid2.groupby("Pais").agg({
    "Vacunaciones diarias por millon": "mean"
    
})
grup1

Vacunaciones diarias por millon
Pais                                              
Afghanistan                             252.795322
Albania                                2110.986900
Algeria                                 448.431373
Andorra                                5740.019704
Angola                                  296.859551
...                                            ...
Wales                                  5544.855469
Wallis and Futuna                      5349.811688
Yemen                                   124.200000
Zambia                                  216.792593
Zimbabwe                               1331.751323

[222 rows x 1 columns]

In [ ]:
v5=grup1["Vacunaciones diarias por millon"].idxmin()
v6=grup1["Vacunaciones diarias por millon"].min()

print("el pais con menor numero de vacunaciones diarias es ",v5," con un promedio de ",v6," por cada millon de habitantes")

el pais con menor numero de vacunaciones diarias es  Bonaire Sint Eustatius and Saba  con un promedio de  0.0  por cada millon de habitantes


Podemos observar que el país Bonaire Sint Eustatius and Saba es el que menos vacunas diarias aplica ya que no ha empezado su proceso de vacunaciones.

Ahora buscaremos la respuesta a la misma pregunta excluyendo a los países que no han empezado a vacunar a su poblacion

In [ ]:
grup2=grup1[grup1["Vacunaciones diarias por millon"] > 0]
grup2

Vacunaciones diarias por millon
Pais                                              
Afghanistan                             252.795322
Albania                                2110.986900
Algeria                                 448.431373
Andorra                                5740.019704
Angola                                  296.859551
...                                            ...
Wales                                  5544.855469
Wallis and Futuna                      5349.811688
Yemen                                   124.200000
Zambia                                  216.792593
Zimbabwe                               1331.751323

[220 rows x 1 columns]

In [ ]:
v7=grup2["Vacunaciones diarias por millon"].idxmin()
v8=grup2["Vacunaciones diarias por millon"].min()
print("El pais con menor numero de vacunaciones diarias es ",v7," con un promedio de ",v8," por cada millon de habitantes")

El pais con menor numero de vacunaciones diarias es  Democratic Republic of Congo  con un promedio de  7.4140625  por cada millon de habitantes


El país con menor promedio de vacunaciones diarias es Democratic Republic of Congo, con apenas ~7.5 vacunaciones diarias por cada millon de habitantes

# **Base de datos secundaria: Graduate Admission 2**

link:https://www.kaggle.com/mohansacharya/graduate-admissions

**Introducción**

Definir la universidad en la que estudiarás tu carrera universitaria es una decisión muy importante en tu vida, ya que necesitas estar bien preparado para obtener un buen lugar en el mercado laboral. Es por ello, que al elegir una universidad, ya sea pública o privada, debe cumplir con tus expectativas profesionales a corto, mediano y largo plazo.

Para que elijas una buena institución, debes comenzar por hacer una investigación exhaustiva que te permita comparar opciones y reflexionar sobre qué universidad es la mejor para ti.

**Objetivos**

Este conjunto de datos se creó con el fin de ayudar a los estudiantes a preseleccionar universidades con sus perfiles. El resultado previsto les da una idea justa sobre sus posibilidades en una universidad concreta.

**Limpieza de datos**

Importamos las librerías las cuales podremos utilizar para la lectura y limpieza de datos. Despues se mandará llamar el archivo el cual esta en formato CVS-Excel


In [ ]:
import pandas as pd
import numpy as np
import json

In [ ]:
df_ap = pd.read_csv("/content/Admission_Predict.csv")

**Visualizando la base de datos**

Con la función head podemos visualizar los primeros 10 datos de la base

In [ ]:
df_ap.head(10)

Serial No.  GRE Score  TOEFL Score  ...  CGPA  Research  Chance of Admit 
0           1        337          118  ...  9.65         1              0.92
1           2        324          107  ...  8.87         1              0.76
2           3        316          104  ...  8.00         1              0.72
3           4        322          110  ...  8.67         1              0.80
4           5        314          103  ...  8.21         0              0.65
5           6        330          115  ...  9.34         1              0.90
6           7        321          109  ...  8.20         1              0.75
7           8        308          101  ...  7.90         0              0.68
8           9        302          102  ...  8.00         0              0.50
9          10        323          108  ...  8.60         0              0.45

[10 rows x 9 columns]

Podemos observar los primeros 10 datos 

Con la función tail podemos observar los ultimos 10 datos de la base 

In [ ]:
df_ap.tail(10)

Serial No.  GRE Score  TOEFL Score  ...  CGPA  Research  Chance of Admit 
390         391        314          102  ...  8.24         0              0.64
391         392        318          106  ...  8.65         0              0.71
392         393        326          112  ...  9.12         1              0.84
393         394        317          104  ...  8.76         0              0.77
394         395        329          111  ...  9.23         1              0.89
395         396        324          110  ...  9.04         1              0.82
396         397        325          107  ...  9.11         1              0.84
397         398        330          116  ...  9.45         1              0.91
398         399        312          103  ...  8.78         0              0.67
399         400        333          117  ...  9.66         1              0.95

[10 rows x 9 columns]

Podemos observar los ultimos 10 datos

Con la función shape podemos observar cuantas columnas y filas hay 

In [ ]:
df_ap.shape

(400, 9)

La base de datos esta conformada por 400 filas y 9 columnas 

Revisando los nombres de las columnas con la función columns

In [ ]:
df_ap.columns

Index(['Serial No.', 'GRE Score', 'TOEFL Score', 'University Rating', 'SOP',
       'LOR ', 'CGPA', 'Research', 'Chance of Admit '],
      dtype='object')

Los nombres de las 9 columnas de nuestra base de datos son:


*   Serial No.
*   GRE Score
*   TOEFL Score 
*   University Rating 
*   SOP
*   LOR
*   CGPA 
*   Research 
*   Chance of Admit 


Con la función dtypes podemos observar el tipo de datos de las columnas

In [ ]:
df_ap.dtypes

Serial No.             int64
GRE Score              int64
TOEFL Score            int64
University Rating      int64
SOP                  float64
LOR                  float64
CGPA                 float64
Research               int64
Chance of Admit      float64
dtype: object

Todas las columnas son del tipo entero a excepción de las columnas SOP, LOR,CGPA y Chance of admit que son del tipo flotantes.

Con la función isnull.sum podemos observar el recuento de las celdas vacías por columnas

In [ ]:
df_ap.isnull().sum()

Serial No.           0
GRE Score            0
TOEFL Score          0
University Rating    0
SOP                  0
LOR                  0
CGPA                 0
Research             0
Chance of Admit      0
dtype: int64

No hay celdas vacías 